src/librustc/lint/builtin.rs

```diff
@@ -318,7 +304,6 @@ impl LintPass for HardwiredLints {
             DEPRECATED,
             UNUSED_UNSAFE,
             UNUSED_MUT,
-            COERCE_NEVER,
```

Comment:
we also need to register that these lints are removed. Like so:

https://github.com/rust-lang/rust/blob/883e74645d350b6752cb94d48f46363f6f8789e9/src/librustc_lint/lib.rs#L316-L317

Presumably something like:

```rust
    store.register_removed("coerce_never",
        "converted into hard error, see https://github.com/rust-lang/rust/issues/XXX");
    store.register_removed("resolve_trait_on_defaulted_unit",
        "converted into hard error, see https://github.com/rust-lang/rust/issues/XXX");
```